resources/Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM ubuntu:18.04 as dev
+
+ARG RUST_TOOLCHAIN="1.40.0"
+ARG CLH_SRC_DIR="/cloud-hypervisor"
+ARG CLH_BUILD_DIR="$CLH_SRC_DIR/build"
+ARG CARGO_REGISTRY_DIR="$CLH_BUILD_DIR/cargo_registry"
+ARG CARGO_GIT_REGISTRY_DIR="$CLH_BUILD_DIR/cargo_git_registry"
+
+ENV CARGO_HOME=/usr/local/rust
+ENV RUSTUP_HOME=/usr/local/rust
+ENV PATH="$PATH:$CARGO_HOME/bin"
+
+# Install all CI dependencies
+RUN apt-get update
+RUN apt-get -yq upgrade
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -yq \
+	build-essential \
+	bc \
+	docker.io \
+	curl \
+	wget \
+	sudo \
+	mtools \
+	libssl-dev \
+	pkg-config \
+	flex \
+	bison \
+	libelf-dev \
+	qemu-utils \
+	qemu-system \
+	libglib2.0-dev \
+	libpixman-1-dev \
+	libseccomp-dev \
+	libcap-ng-dev \
+	socat \
+	dosfstools \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install the rust toolchain
+RUN nohup curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain "$RUST_TOOLCHAIN" \
+    && cargo install cargo-audit \
```

Comment:
Cool. Does this container already have the tools installed in it: e.g. clippy?

Reply:
The toolchain install pulls clippy and rustfmt, but I will expliclty add them in the Dockerfile.

Reply:
Can you also modify the `run_cargo_tests.sh` script to do "which cargo-audit || cargo install cargo-audit` rather than the current `--force`

Reply:
Done